cachix/src/Cachix/Deploy/StdinProcess.hs

```diff
@@ -19,7 +19,3 @@ readProcess cmd args input = do
         }
   hPutStr stdin input
   hClose stdin
-  exitcode <- waitForProcess ph
```

Comment:
I don't quite understand how locking works if it's not waiting for the process to finish?

Reply:
This is, admittedly, a hack to consume the deployment messages eagerly, instead of accumulating them. While waiting for the binary to exit, the agent would continue to add deployments to the backlog. This can include duplicate deployments if the `startDeployment` message takes a while to be delivered.

Locking is done in the deployment binary (because the agent might be upgraded and lose the lock).

Waiting for the process is the right thing to do. With the current backend setup, I was thinking of tracking whether we're running a deployment and dropping any incoming deployments until the current one exits. But if the agent is restarted, we'll lose that state and we're back to this again.